airflow/jobs.py

```diff
@@ -191,11 +198,16 @@ def run(self):
         self.id = id_
 
         # Run
-        self._execute()
+        try:
```

Comment:
Also this is the suggestion from @mistercrunch 
https://github.com/apache/incubator-airflow/pull/1122#issuecomment-193252883